### ISSUE TITLE
SchemaDiff throws `SchemaException("There is no column ....")` since v2.5.5

### ISSUE THREAD

**Issue by arima-ryunosuke** [doctrine/dbal#2501]
code:

```
<?php
require_once __DIR__ . '/vendor/autoload.php';

$common = array(
    'driver'   => 'pdo_mysql',
    'host'     => '127.0.0.1',
    'user'     => 'user',
    'password' => 'pass',
);

// from Connection
$fromParams = $common;
$fromParams['dbname'] = 'test1';
$fromConn = \Doctrine\DBAL\DriverManager::getConnection($fromParams);

// to Connection
$toParams = $common;
$toParams['dbname'] = 'test2';
$toConn = \Doctrine\DBAL\DriverManager::getConnection($toParams);

// create dummy table
$fromConn->exec('
CREATE TABLE IF NOT EXISTS t_table (
  pid int(11) NOT NULL,
  PRIMARY KEY (pid)
)');
$toConn->exec('
CREATE TABLE IF NOT EXISTS t_table (
  sid int(11) NOT NULL,
  PRIMARY KEY (sid)
)');

// print sql diff
$fromSchema = $fromConn->getSchemaManager()->createSchema();
$toSchema = $toConn->getSchemaManager()->createSchema();
$diff = \Doctrine\DBAL\Schema\Comparator::compareSchemas($fromSchema, $toSchema);
print_r($diff->toSql($fromConn->getDatabasePlatform()));
```

v2.5.4 is

```
[0] => ALTER TABLE t_table DROP PRIMARY KEY
[1] => ALTER TABLE t_table CHANGE pid sid INT NOT NULL
[2] => ALTER TABLE t_table ADD PRIMARY KEY (sid)
```

but v2.5.5 is

```
Fatal error: Uncaught exception 'Doctrine\DBAL\Schema\SchemaException' with message 'There is no column with name 'sid' on table 't_table'.' in /path/lib/Doctrine/DBAL/Schema/SchemaException.php on line 86
```

I would like to append to a "MySqlPlatform.php:702" like a below.

```
if (!$diff->fromTable->hasColumn($columnName)) {
    continue;
}
```

refs #2302

**Comment by Seb33300** [comment]
Same issue here!

**Comment by webdevilopers** [comment]
Same here using `doctrine:migrations:diff`. Every **@ManyToMany** Join Table generates:

> [Doctrine\DBAL\Schema\SchemaException]                         
  There is no column with name 'id' on table '...'.

```php
    /**
     * @ORM\ManyToMany(targetEntity="AppBundle\Entity\Contract", inversedBy="expectedFaults")
     * @ORM\JoinTable(name="expected_faults",
     *      joinColumns={@ORM\JoinColumn(name="fault_id", referencedColumnName="id")},
     *      inverseJoinColumns={@ORM\JoinColumn(name="contract_id", referencedColumnName="id")}
     *      )
     */
```

**Comment by NikDevPHP** [comment]
+1

**Comment by weaverryan** [comment]
I can confirm this. In fact, we have a number of users that have been hitting this in our ManyToMany tutorial (https://knpuniversity.com/screencast/collections/many-to-many-extra-fields#comment-3094129612), so it's definitely legit (but I know these things are complex!).

**Comment by ste93cry** [comment]
I hit the bug too, and can confirm that the fix provided by @arima-ryunosuke works in my case, which is adding new columsn to the table and then recreate the primary key to use the old columns plus the new ones.

**Comment by arima-ryunosuke** [comment]
This issue is fixed at v2.5.13 by #2696
